README.md

```diff
@@ -1,5 +1,109 @@
+- Performing updates to the Jenkins Core system to support any additional event-listeners needed, for example, for acquiring audit information.
+- Creation of new audit events, with the provision of customizable interfaces for user-defined configuration. For example;
+    - Enable users to configure how audit logs are stored and used.
+    - Provide admin settings to configure a relational database or local directory to write log files into.
+    - TODO: Add some other use cases
+- TODO: Add more use cases and applications of the plugin
+
+Getting Started: Prerequisites
+---------------
+
++ Make sure you have a [JIRA account](https://issues.apache.org/jira/).
++ Make sure you have a [GitHub account](https://github.com/signup/free).
++ If you're planning to implement a new feature, it makes sense to discuss your changes
+  on [Gitter](https://gitter.im/jenkinsci/outreachy) first.
+  This way you can make sure you're not wasting your time on something that isn't
+  considered to be in the audit-logging project's scope.
++ Submit a ticket for your issue, assuming one does not already exist.
+  + Clearly describe the issue, including steps to reproduce when it is a bug.
+  + Make sure you fill in the earliest version that you know has the issue.
++ Fork the [repository][project-repo] on GitHub, take a look at the [project's JIRA page][project-jira] and [Wiki]([audit-log-wiki]).
+
+Getting Started: Setup
+----------------------
+
+#### TODO - how to setup/start the plugin with Jenkins, etc
+
+
+
+Usage
+----------------------
+
+#### TODO - demo, screenshot, gifs of running audit logging, etc
+
+
+Contributing to the Audit Log Project
+--------------------------
+
+We accept Pull Requests via GitHub. The [gitter](https://gitter.im/jenkinsci/outreachy) is the
+main channel of communication for contributors.
+
+Before you dig right into the code, there are a few guidelines that we need
+contributors to follow so that we can have a chance of keeping on top of
+things and it makes applying PRs easier for us:
++ Create a topic branch from where you want to base your work (this is usually the master branch).
+  Push your changes to a topic branch in your fork of the repository.
++ Make commits of logical units.
++ Respect the original code style: by using the same [codestyle][code-style],
+  patches should only highlight the actual difference, not being disturbed by any formatting issues:
+  + Only use spaces for indentation.
+  + Create minimal diffs - disable on save actions like reformat source code or organize imports.
+    If you feel the source code should be reformatted, create a separate PR for this change.
+  + Check for unnecessary whitespace with `git diff --check` before committing.
++ Make sure your commit messages are in the proper format. Your commit message should contain the key of the JIRA issue.
+```
+[MCOMPILER-XXX] - Subject of the JIRA Ticket
+ Optional supplemental description.
```

Comment:
Should add a blank line between the "subject" line and the "body" of the commit message.